roles/neutron-common/templates/usr/local/bin/neutron-reset

```diff
@@ -39,12 +40,14 @@ class Interface(object):
 
         pattern = re.compile(r'^\d+: ([^:]+): ')
 
+        iface_list = list()
         for line in output.splitlines():
             match = pattern.match(line)
             if match:
                 iface = cls(match.group(1), namespace=namespace)
                 if iface.is_neutron_interface():
-                    yield iface
+                    iface_list.append(iface)
+        return iface_list
```

Comment:
Interface class needs to remember the previous state of the devices before devices are being reset, meaning which tap/tun device is attached to which brq device. The generator would not allow it to do that.